sql/item.cc

```diff
@@ -6336,9 +6387,18 @@ bool Item_field::fix_fields(THD *thd, Item **reference)
   Field *from_field= (Field *)not_found_field;
   bool outer_fixed= false;
   SELECT_LEX *select;
-  if (context)
+  Name_resolution_context *ctx= context;
+
+  if (depended_from)        // this is outer reference and we have fixed before
+  {
+    if (!alias_name_used)   // short circuit fix_outer_refs if this wasn't alias
+      ctx= &depended_from->context;
```

Comment:
If the field was resolved as an alias in the select list, we cannot bypass resolving here again, otherwise we can.